File: lib/models/song.dart
class Song {
  final String id;
  final String title;
  final String artist;
  final String album;
  final String path;
  final String? coverPath;

  Song({
    required this.id,
    required this.title,
    required this.artist,
    required this.album,
    required this.path,
    this.coverPath,
  });

  factory Song.fromJson(Map<String, dynamic> json) {
    return Song(
      id: json['id'],
      title: json['title'],
      artist: json['artist'],
      album: json['album'],
      path: json['path'],
      coverPath: json['coverPath'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'title': title,
      'artist': artist,
      'album': album,
      'path': path,
      'coverPath': coverPath,
    };
  }
}

File: src/models/Song.js
export default class Song {
    constructor(id, title, artist, album, duration, path, coverPath) {
        this.id = id;
        this.title = title;
        this.artist = artist;
        this.album = album;
        this.duration = duration;
        this.path = path;
        this.coverPath = coverPath || null;
        this.isPlaying = false;
        this.progress = 0;
    }

    static fromJson(json) {
        return new Song(
            json.id,
            json.title,
            json.artist,
            json.album,
            json.duration,
            json.path,
            json.coverPath
        );
    }

    toJson() {
        return {
            id: this.id,
            title: this.title,
            artist: this.artist,
            album: this.album,
            duration: this.duration,
            path: this.path,
            coverPath: this.coverPath
        };
    }
}
